app/src/main/java/com/lagradost/cloudstream3/extractors/VidSrcTo.kt

```diff
@@ -0,0 +1,74 @@
+package com.lagradost.cloudstream3.extractors
+
+import android.util.Base64
+import android.util.Log
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.lagradost.cloudstream3.SubtitleFile
+import com.lagradost.cloudstream3.apmap
+import com.lagradost.cloudstream3.app
+import com.lagradost.cloudstream3.utils.ExtractorApi
+import com.lagradost.cloudstream3.utils.ExtractorLink
+import java.net.URLDecoder
+import javax.crypto.Cipher
+import javax.crypto.spec.SecretKeySpec
+
+class VidSrcTo : ExtractorApi() {
+    override val name = "VidSrcTo"
+    override val mainUrl = "https://vidsrc.to"
+    override val requiresReferer = true
+
+    override suspend fun getUrl(
+            url: String,
+            referer: String?,
+            subtitleCallback: (SubtitleFile) -> Unit,
+            callback: (ExtractorLink) -> Unit
+    ) {
+        val mediaId =
+                app.get(url).document.selectFirst("ul.episodes li a")?.attr("data-id") ?: return
+        val res =
+                app.get("$mainUrl/ajax/embed/episode/$mediaId/sources")
+                        .parsedSafe<VidsrctoEpisodeSources>()
+        if (res?.status == 200) {

```

Comment:
if (res.status != 200) return

Reply:
The core idea is to make it as flat as possible, as that is easier to read then an pyramid of indentation

Reply:
https://en.wikipedia.org/wiki/Pyramid_of_doom_(programming)